Task2/aga/TicTacToe/TicTacToe/TheGame.xaml.cs

```diff
@@ -0,0 +1,242 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+
+namespace TicTacToe
+{
+    /// <summary>
+    /// Interaction logic for TheGame.xaml
+    /// </summary>
+    public partial class TheGame : Window
+    {
+        public TheGame()
+        {
+            InitializeComponent();
+
+        }
+        Click click = new Click();
+        private bool[] clickedButtons = new bool[9];
+        private object winner = "";
+
+        private void A1_Click(object sender, RoutedEventArgs e)
+        {
+            Button b = (Button)sender;            
+            int turn = click.GetTurn();
```

Comment:
Using a Control class (Button in this case) inside a C# code should be kept to a minimum. If you use ViewModel and bind to a Command instead of doing everything in code-behind you will avoid having to cast objects to Buttons.